src/Evolution/Systems/GeneralizedHarmonic/Equations.cpp

```diff
@@ -248,9 +249,71 @@ void ComputeDuDt<Dim>::apply(
     }
   }
 }
+
+template <size_t Dim>
+void ComputeNormalDotFluxes<Dim>::apply(
+    const gsl::not_null<tnsr::aa<DataVector, Dim>*>
+        spacetime_metric_normal_dot_flux,
+    const gsl::not_null<tnsr::aa<DataVector, Dim>*> pi_normal_dot_flux,
+    const gsl::not_null<tnsr::iaa<DataVector, Dim>*> phi_normal_dot_flux,
+    const tnsr::aa<DataVector, Dim>& spacetime_metric,
+    const tnsr::aa<DataVector, Dim>& pi, const tnsr::iaa<DataVector, Dim>& phi,
+    const Scalar<DataVector>& gamma1, const Scalar<DataVector>& gamma2,
+    const Scalar<DataVector>& lapse, const tnsr::I<DataVector, Dim>& shift,
+    const tnsr::II<DataVector, Dim>& inverse_spatial_metric,
+    const tnsr::i<DataVector, Dim>& unit_normal) noexcept {
+  const auto shift_dot_normal = get(dot_product(shift, unit_normal));
+
+  auto upper_normal_dot_phi =
```

Comment:
I missed what the meaning of "upper" was?

Reply:
The normal is a lower index tensor, so this is the "upper normal" in that the index is raised. I can just change it to normal dot phi since there's only one way that can make sense.